homeassistant/components/device_tracker/tado.py

```diff
@@ -68,22 +68,18 @@ def __init__(self, hass, config):
         self.websession = async_create_clientsession(
             hass, cookie_jar=aiohttp.CookieJar(unsafe=True, loop=hass.loop))
 
-        self.success_init = self._update_info()
+        self.success_init = asyncio.run_coroutine_threadsafe(
```

Comment:
I think we should make a clearer pattern than this. Could we await directly after instantiating the scanner instead from `async_get_scanner` and make `_async_update_info` non private?

Reply:
Drop `success_init` and just return `None` if `async_update_info` returns false.

Reply:
Well the problem is that this pattern is well established. We can explore changing it, but let's do that in another PR. This one I would like to include in a hotfix